src/templates/components/image/index.js

```diff
@@ -1,33 +1,61 @@
 import { h } from 'preact';
 
-const ImageSource=({fileName, data}) => {
-	return <source srcset={`${fileName.split('.')[0]}${data.sizeID}.${data.format}${data.query}`} media={`(min-width:${data.minWidth}px)`} />
-}
-
+/**
+ * Responsive image component, using the <picture> element
+ * @param {string} alt - Image alt tag
+ * @param {string} decoding='async' - 'async', 'sync', 'auto'
+ * @param {string} loading='lazy' - 'lazy', 'eager'
+ * @param {string} src - Reference to local/relative or remote image URI
+ * @param {string} className - Class name
+ * @param {array} [sources]	- Array of Objects for <picture> —> <source> attributes {src, media}
+ **/
 const Image=({
-	componentClass,
-	imageClass,
+	className,
 	alt,
-	fileName,
-	altSizes,
+	src,
+	sources,
 	decoding='async',
 	loading='lazy'
-	}) => 
+	}) =>
 
-<picture class={`image ${componentClass ? componentClass : ''}`}>
+<picture class={`${className ? className : ''}`}>
 
 	{
-		altSizes && altSizes.map(size => ImageSource({fileName:fileName, data:size}))
+		sources && sources.map(srcData => <source srcset={srcData.src} media={srcData.media} />)
 	}
 
 	<img 
 		alt={alt}
-		class={`${imageClass ? imageClass : ''}`}
+		class={`${className ? className : ''}`}
 		decoding={decoding}
-		src={`${fileName}`}
+		src={`${src}`}
 		loading={loading}
 	/>
 
 </picture>
+export default Image;
+
+/**
+ * Wrapper for ImageBlock component, usually where a caption is required
+ * @param {Object} children - an ImageBlock 
+ * @param {string} caption - Text to display with image
+ * @param {string} className - Class name 
+ */
+export const Figure = ({
+	children,
+	className,
+	caption}) =>
+
+<figure class={`${className ? className : ''}`} >
```

Comment:
Do we need to have two classNames here, one for the figcaption and one for the figure?  Just in case we need to target them differently without using the tagname?  Or maybe we could use the same one but append something to them to differentiate?  perhaps __wrap and __caption?

Reply:
I was thinking the same over the weekend. My memory of the team discussion over this lent towards removing appended strings, so I thought to add a second class. 
className will attach to the figure, and captionClassName (?) for the caption